test/csv_test.exs

```diff
@@ -8,9 +8,9 @@ defmodule CSVTest do
 
   test "decodes" do
     stream = Stream.map(["a,be", "c,d"], &(&1))
-    result = CSV.decode(stream) |> Enum.into([]) |> Enum.sort
 
-    assert result == [~w(a be), ~w(c d)]
+    assert (CSV.decode!(stream) |> Enum.into([]) |> Enum.sort) == [~w(a be), ~w(c d)]
+    assert (CSV.decode(stream) |> Enum.into([]) |> Enum.sort) == [ ok: ~w(a be), ok: ~w(c d) ]
```

Comment:
Can we separate the tests for the `decode!` from the `decode` API? I think this would make the tests much clearer